File: TestApp/app/src/main/java/com/romiope/testapp/database/ServersDao.java
package com.romiope.testapp.database;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import com.romiope.testapp.network.entities.ServerResponse;

import java.util.List;

import static android.arch.persistence.room.OnConflictStrategy.REPLACE;

@Dao
public interface ServersDao {

    @Query("SELECT * FROM ServerResponse")
    List<ServerResponse> getAllServers();

    @Insert(onConflict = REPLACE)
    void insertServers(List<ServerResponse> server);

    @Query("DELETE FROM ServerResponse")
    void clean();
}

File: app/src/main/java/com/example/android_party/data/ServersDao.kt
package com.example.android_party.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface ServersDao {
    @get:Query("SELECT * FROM server")
    val getAll: List<Server>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(servers: List<Server>)

    @Query("DELETE FROM server")
    fun delete()
}